src/pages/Video/Video.js

```diff
@@ -19,6 +24,11 @@ const Video = () => {
   const location = useLocation();
   const navigate = useNavigate();
 
+  const { mutate: addToHistory } = useAddToHistory();
+  useEffect(() => {
+    if (video && isAuth) addToHistory(video);
+  }, [addToHistory, isAuth, video]);
```

Comment:
the dependency array should be empty, and then it should call a function like dispatch and there check whether or not the the user is authenticated

Reply:
yes that is what i did at first. but the way i am fetching `video` here, at the initial render, i do not have access to `video` object so had to include `video` in the dependency array. Had to include `addToHistory` in the array because eslint complains, but i think its fine to include it as its reference won't change across renders. even if multiple requests are fired from here, i have changed the backend so that each video as only one latest entry in users history. i have checked a few times this fires only once when user is authenticated and video is available so i didn't worry much about this. thanks